src/spec/metrics/acceptance_test.go

```diff
@@ -80,56 +67,104 @@ var _ = Describe("Metrics are received", func() {
+					instances, err := bosh.Instances(deployment, bosh.MatchByIndexedName("mysql/0"))
+					Expect(err).NotTo(HaveOccurred())
+					Expect(instances).To(HaveLen(1))
+
+					promQL := `_` + strings.ReplaceAll(SourceID, "-", "_") + `_system_ephemeral_disk_used_percent{source_id="` + SourceID + `",index="` + instances[0].Id() + `"}`
 
-					initialDiskUsePercent = extractIntMatchingRegex(ephDiskMetricsOutput, ephemeralDiskUsedMetricRegex)
-					session.Terminate()
+					var query Query
+					Eventually(func() int {
+						session := cf.Cf("query", promQL).Wait(10 * time.Second)
+
+						err := json.Unmarshal(session.Out.Contents(), &query)
+						Expect(err).NotTo(HaveOccurred())
+
+						if query.Data.Result == nil {
+							return 0
+						}
+
+						return query.ValueAsInt()
+					}, "60s", "5s").Should(BeNumerically(">=", 0))
+
+					initialDiskUsePercent = query.ValueAsInt()
 				})
 			})
 
 			By("Writing 50% of available ephemeral disk", func() {
 				amountToAllocate := availBytes / 2
 				vmCommand := fmt.Sprintf("sudo fallocate -l%d /var/vcap/data/file", amountToAllocate)
-				args = []string{"ssh", "mysql/0", "-c", vmCommand}
-				session = testhelpers.ExecuteBosh(args, 10*time.Second)
-				Expect(session.ExitCode()).To(BeZero())
+				_, err := bosh.RemoteCommand(deployment, "mysql/0", vmCommand)
+				Expect(err).NotTo(HaveOccurred())
 			})
 
 			By("Capturing the ephemeral disk usage metric after allocating a file", func() {
 				workflowhelpers.AsUser(TestSetup.AdminUserContext(), time.Microsecond, func() {
-					session := cf.Cf("tail", "-f", SourceID)
-					Eventually(session.Out, 40*time.Second).Should(gbytes.Say(ephemeralDiskUsedMetricRegex))
-					out := session.Out.Contents()
-					finalDiskUsePercent = extractIntMatchingRegex(out, ephemeralDiskUsedMetricRegex)
-					session.Terminate()
-				})
-			})
+					instances, err := bosh.Instances(deployment, bosh.MatchByIndexedName("mysql/0"))
+					Expect(err).NotTo(HaveOccurred())
+					Expect(instances).To(HaveLen(1))
 
-			By("Having measured a significant change in ephemeral disk usage", func() {
-				emittedDiskUsageChange := finalDiskUsePercent - initialDiskUsePercent
-				fmt.Printf("emittedDiskUsageChange: %d", emittedDiskUsageChange)
-				fmt.Printf("expectedDiskUsageChange: %d", expectedDiskUsageChange)
-				Expect(emittedDiskUsageChange).To(
-					BeNumerically(">=", expectedDiskUsageChange-2),
-					`Expected the increase in percent ephemeral disk used to close to or greater than the expected percent to write.
+					promQL := `_` + strings.ReplaceAll(SourceID, "-", "_") + `_system_ephemeral_disk_used_percent{source_id="` + SourceID + `",index="` + instances[0].Id() + `"}`
+
+					var query Query
+					Eventually(func() int {
+						session := cf.Cf("query", promQL).Wait(10 * time.Second)
+
+						err := json.Unmarshal(session.Out.Contents(), &query)
+						Expect(err).NotTo(HaveOccurred())
+
+						if query.Data.Result == nil {
+							return 0
+						}
```

Comment:
This metric code repeats twice and may make sense to pull out into a helper.

```go
func queryEphemeralDiskUsedPercentMetric(instanceGUID string) (percentUsed int64, err error) {
	metricName  := "/" + SourceID + "/system/ephemeral_disk_used_percent"
	promqlQuery := fmt.Sprintf(`%s{source_id=%q,index=%q}`, SanitizeMetricName(metricName), SourceID, instanceGUID)
	var result struct {
		Data struct {
			Result []struct {
				Value []json.Number `json:"value"`
			} `json:"result"`
		} `json:"data"`
	}

	session := cf.Cf("query", promqlQuery).Wait(10 * time.Second)

	Expect(json.NewDecoder(session.Out).Decode(&result)).To(Succeed())
        /* ...consisistency checks... */
        return result.Data.Result[0].Value[1].Int64()
}
```

Reply:
created new helper function to better isolate the `cf query` logic